go.sum

```diff
@@ -455,6 +455,8 @@ github.com/xiang90/probing v0.0.0-20190116061207-43a291ad63a2/go.mod h1:UETIi67q
 github.com/xordataexchange/crypt v0.0.3-0.20170626215501-b2862e3d0a77/go.mod h1:aYKd//L2LvnjZzWKhF00oedf4jCCReLcmhLdhm1A27Q=
 github.com/yookoala/realpath v1.0.0/go.mod h1:gJJMA9wuX7AcqLy1+ffPatSCySA1FQ2S8Ya9AIoYBpE=
 github.com/yuin/goldmark v1.2.1/go.mod h1:3hX8gzYuyVAZsxl0MRgGTJEmQBFcNTphYh9decYSb74=
+go.etcd.io/bbolt v1.3.2/go.mod h1:IbVyRI1SCnLcuJnV2u8VeU0CEYM7e686BmAb1XKL+uU=
+go.etcd.io/bbolt v1.3.3/go.mod h1:IbVyRI1SCnLcuJnV2u8VeU0CEYM7e686BmAb1XKL+uU=
```

Comment:
Why do the version 1.3.2 and 1.3.3 appear here?

Reply:
In short, Golang will scan and validate the `go.mod` file of all versions that meets and write into `go.sum`

---

Detailed description here: we can get all modules that depend on `bbolt`:

```shell
:) go mod graph | grep go.etcd.io/bbolt
github.com/tikv/pd go.etcd.io/bbolt@v1.3.5
go.etcd.io/bbolt@v1.3.5 golang.org/x/sys@v0.0.0-20200202164722-d101bd2416d5
go.etcd.io/etcd@v0.5.0-alpha.5.0.20191023171146-3cf2f69b5738 go.etcd.io/bbolt@v1.3.3
go.etcd.io/etcd@v0.0.0-20191023171146-3cf2f69b5738 go.etcd.io/bbolt@v1.3.3
github.com/spf13/viper@v1.4.0 go.etcd.io/bbolt@v1.3.2
```

So, `v1.3.2`, `v1.3.3` and `v1.3.5` will all be write into `go.sum`.

According to the design decision, please refer to [Go Modules Reference](https://golang.org/ref/mod#go-sum-files).